share/spack/win-compiler-wrapper/cl.cxx

```diff
@@ -0,0 +1,406 @@
+#include "cl.hxx"
+
+// String helper methods adding cxx20 features to cxx14
+bool startswith(std::string &arg, std::string& match)
+{
+    size_t matchLen = match.size();
+    if (matchLen > arg.size())
+        return false;
+    return arg.compare(0, matchLen, match) == 0;
+}
+bool startswith(std::string &arg, const char * match){
+    return startswith(arg, (std::string)match);
+}
+
+bool endswith(std::string &arg, std::string &match)
+{
+    size_t matchLen = match.size();
+    if (matchLen > arg.size())
+        return false;
+    return arg.compare(arg.size() - matchLen, matchLen, 0) == 0;
```

Comment:
I don't see how this works without referring to `match` (just `matchLen`).

Reply:
Refactor error.